src/java.base/share/classes/java/nio/file/Path.java

```diff
@@ -248,6 +256,60 @@ public static Path of(URI uri) {
      */
     Path getFileName();
 
+    /**
+     * Returns the file name extension of this path as a {@code String}. The
+     * extension is defined to be the portion of the {@code String}
+     * representation of the file name after the last dot ('.'). All leading
+     * dots are ignored. If the extension is missing, then an
+     * {@link String#isEmpty() empty} {@code String} ({@code ""}) is returned.
+     * This will occur if the path has zero elements ({@link #getFileName()}
+     * returns {@code null}), the file name string does not contain a dot, or
+     * only the first character is a dot. If the last character is a dot, the
+     * extension is the empty string.
+     *
+     * @implSpec
+     * The default implementation is equivalent for this path to:
+     * <pre>{@code
+     * int lastDot = fileName.lastIndexOf('.');
+     * if (lastDot <= 0)
+     *     return null;
+     * return lastDot == fileName.length() - 1 ?
+     *     "" :
+     *     fileName.substring(lastDot + 1);
+     * }</pre>
+     *
+     * @return  the file name extension of this path, which might be the
+     *          empty string, or {@code null} if no extension is found
+     *
+     * @since 20
+     */
+    default String getExtension() {
+        Path fileName = getFileName();
+        if (fileName == null)
+            return "";
+
+        String fileNameString = fileName.toString();
+        int length = fileNameString.length();
+
+        // A file name string of length unity has a null extension
+        if (length > 2) {
+            int lastDotIndex = fileNameString.lastIndexOf('.');
+            // Indeterminate if no dot or only at first index
+            if (lastDotIndex > 0) {
+                return lastDotIndex == length - 1 ?
+                    "" : // empty string if last dot at last index
+                    fileNameString.substring(lastDotIndex + 1);
+            }
+        } else if (length == 2 &&
+                   !(fileNameString.charAt(0) == '.' ||
+                     fileNameString.charAt(1) != '.')) {
```

Comment:
I would have coded it as:
```
   (length == 2 &&
      fileNameString.charAt(0) != '.' &&
      fileNameString.charat(1) == '.')) {
```
As a better match for the comment.

Reply:
I had it that way originally but changed it thinking that fewer boolean expressions might need to be evaluated.

Reply:
Changed in 3f510ecc32089589203b1fffb13eedc579aaa8bb.